File: learn_android/code/JAVA/ViewModeltest/app/src/main/java/com/hkc/viewmodeltest/MainActivity.java
package com.hkc.viewmodeltest;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    MainModel mainModel;
    TextView textView;
    Button button1,button2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mainModel = new ViewModelProvider(this).get(MainModel.class);
        textView = findViewById(R.id.textView);
        button1 = findViewById(R.id.button1);
        button2 = findViewById(R.id.button2);

        button1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                mainModel.number ++;
                textView.setText(String.valueOf(mainModel.number));
            }
        });

        button2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                mainModel.number += 2;
                textView.setText(String.valueOf(mainModel.number));
            }
        });

    }
}

File: learn_android/code/KotlinCoroutine/app/src/main/java/com/hkc/kotlincoroutine/MainActivity.kt
package com.hkc.kotlincoroutine

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    lateinit var textview:TextView
    lateinit var button: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        textview = findViewById<TextView>(R.id.textview)


        textview.text = "Jack"

        button = findViewById<Button>(R.id.button).also {
            it.setOnClickListener {

            }
        }


    }
}